package/MDAnalysis/lib/distances.py

```diff
@@ -1432,25 +1493,62 @@ def calc_bonds(coords1: Union[np.ndarray, 'AtomGroup'],
     .. versionchanged:: 2.3.0
        Can now accept an :class:`~MDAnalysis.core.groups.AtomGroup` as an
        argument in any position and checks inputs using type hinting.
+    .. versionchanged:: 2.4.0
+       Can now optionally use the fast distance functions from distopia
     """
     numatom = coords1.shape[0]
     bondlengths = _check_result_array(result, (numatom,))
 
     if numatom > 0:
         if box is not None:
             boxtype, box = check_box(box)
-            if boxtype == 'ortho':
-                _run("calc_bond_distance_ortho",
-                     args=(coords1, coords2, box, bondlengths),
-                     backend=backend)
+            if boxtype == "ortho":
+                if backend == "distopia":
+                    # need explicit branch on backend to prepare input types correctly
+                    # must assign to result to get correct reference on memview
+                    bondlengths = _run(
+                        "calc_bonds_ortho_float",
+                        args=(coords1, coords2, box[:3]),
+                        kwargs={"results": bondlengths.astype(np.float32)},
+                        backend=backend,
+                    )
+                    # upcast is currently required, change for 3.0
+                    bondlengths = bondlengths.astype(np.float64)
+                else:
+                    _run(
+                        "calc_bond_distance_ortho",
+                        args=(coords1, coords2, box, bondlengths),
+                        backend=backend,
+                    )
             else:
-                _run("calc_bond_distance_triclinic",
-                     args=(coords1, coords2, box, bondlengths),
-                     backend=backend)
+                if backend == "distopia":
+                    # distopia does not support triclinic boxes
```

Comment:
Since it's currently an active choice, should we pass a warning here? (I'd be against a warning if it was the default choice, but I think if folks are trying it out / benchmarking they might want to know that they shouldn't be expecting faster performance for this case? I'm happy if the answer is no here.

Reply:
Yep i'll add a warning.